pbincli/utils.py

```diff
@@ -23,11 +23,16 @@ def check_writable(f):
 
 
 def decompress(s):
-    return zlib.decompress(bytearray(map(ord, b64decode(s.encode('utf-8')).decode('utf-8'))), -zlib.MAX_WBITS)
+     return zlib.decompress(s, -zlib.MAX_WBITS)
 
 
 def compress(s):
-    co = zlib.compressobj(wbits=-zlib.MAX_WBITS)
-    b = co.compress(s) + co.flush()
+     # using compressobj as compress doesn't let us specify wbits
+     # needed to get the raw stream without headers
+     co = zlib.compressobj(wbits=-zlib.MAX_WBITS)
+     return co.compress(s) + co.flush()
 
-    return b64encode(''.join(map(chr, b)).encode('utf-8'))
+
+
+def json_encode(d):
+    return json.dumps(d, separators=(',',':')).encode()
```

Comment:
Are we really need such compact json?
Since 3.4 used `(',', ': ')` as default.
https://docs.python.org/3/library/json.html#json.dump

Reply:
In most places this isn't strictly required, except for the adata. In the v2 format, settings of the paste that are OK to expose to readers of the paste were moved into the adata field, as that is included in the verification of the cipher message, but stored plain text next to the message. Gives us the benefit that readers can be sure that the message wasn't manipulated on the server.

For this to work, both the creator and the reader of a paste need to generate the same adata JSON string from the paste object. This is why I am using arrays with fixed positions instead of objects in the adata section, since JSON doesn't require parsers to preserve the order of properties, but the order in arrays is fixed.

Except that python by default produces human readable JSON with spaces and you [had previously replaced these with a search-and-replace](https://github.com/r4sas/PBinCLI/blob/1ff6e721c7c562168839d1d029224608a5c4f28e/pbincli/actions.py#L44). When I found the above way, I switched to this as it seemed simpler. In the other uses of `json.dumps` we could skip that bit, but then again I still have to encode the result, we have the library function anyway and even conserve a few bytes sending the data.